src/js/components/dialog.js

```diff
@@ -332,8 +332,15 @@ export default class Dialog extends Component {
        ***********************************************************************/
 
       _clickOriginatedInsideDialogOrTrigger (event) {
-        return this.element.contains(event.target) ||
-               event.target.closest(this.triggerSelector)
+
+        // This method uses event.path.some() to walk up the DOM tree to
+        // determine if the event target was within the dialog instead of
+        // this.element.contains(). It does so to prevent this method from
+        // returning a false negative if a click event originating inside
+        // the dialog removed its target from the DOM.
+
+        return event.target.closest(this.triggerSelector) ||
+               event.path.some(el => el.dataset && 'rvtDialog' in el.dataset)
```

Comment:
This looks good and fixes the bug given the same test case attached to the original issue.

When I test it locally I get a console warning that `Event.path` is deprecated and will be removed with guidance to move to the replacement `Event.composedPath()`.

<img width="875" alt="image" src="https://user-images.githubusercontent.com/5590034/188167610-504f1708-dbca-44b6-8ca2-b01eec3548cd.png">


I tested this locally and `event.composedPath().some(...)` worked as expected. Let's go ahead and update this to use the new composedPath method instead.